src/datascience-ui/image-renderer/index.ts

```diff
@@ -0,0 +1,363 @@
+        const needsBackground = customMetadata?.needs_background || 'light';
+        const backgroundColor = needsBackground === 'light' ? 'white' : 'black';
+        const container = document.createElement('div');
+        container.className = 'display';
+        container.style.overflow = 'scroll'; // `overflow:scroll` is the default style used by Jupyter lab.
+
+        const savePlotButton = createSaveAsButton(outputItem, ctx)
+        const plotViewerButton = output.metadata.__displayOpenPlotIcon === true ? createPlotViewerButton(outputItem, ctx) : undefined;
+        const onMouseOver = () => {
+            if (plotViewerButton) {
+                plotViewerButton.className = 'plotIcon';
+            }
+            savePlotButton.className = 'plotIcon';
+        };
+        const onMouseOut = () => {
+            if (plotViewerButton) {
+                plotViewerButton.className = 'plotIcon hidden';
+            }
+            savePlotButton.className = 'plotIcon hidden';
+        };
+        savePlotButton.onmouseover = onMouseOver;
+        if (plotViewerButton) {
+            plotViewerButton.onmouseover = onMouseOver;
+        }
+        element.appendChild(container);
+        container.appendChild(savePlotButton)
+        if (plotViewerButton) {
+            container.appendChild(plotViewerButton)
+        }
+        const imgSrc = outputItem.mime.toLowerCase().includes('svg') && typeof data === 'string' ? undefined : URL.createObjectURL(data);
+        if (imgSrc) {
+            const img = document.createElement('img');
+            img.onmouseover = onMouseOver;
+            img.onmouseout = onMouseOut;
+            img.src = imgSrc;
+            img.style.backgroundColor = backgroundColor;
+            container.appendChild(img);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const imageMetadata = (customMetadata || {})[outputItem.mime] as Record<string, any> | undefined;
+            if (imageMetadata?.height) {
+                img.height = imageMetadata.height;
+            }
+            if (imageMetadata?.width) {
+                img.width = imageMetadata.width;
+            }
+            if (imageMetadata?.unconfined === true) {
+                img.style.maxWidth = 'none';
+            }
+        }
+        else {
+            const div = document.createElement('div');
+            div.onmouseover = onMouseOver;
+            div.onmouseout = onMouseOut;
+            div.style.backgroundColor = backgroundColor;
+            div.className = 'svgContent';
+            div.innerHTML = data as string;
+            container.appendChild(div);
+        }
+    } catch (ex) {
+        console.error(`Failed to render mime type ${mimeString}`, ex);
+    }
+}
+
+function createSaveAsButton(outputItem: OutputItem, ctx: RendererContext<unknown>) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const btn = document.createElement('button') as any;
```

Comment:
Using pure JS,  no need of React.
I couldn't use react as we use webpack 4 and we cannot bundle react projects into ESM targets.
It was either upgrade to webpack 5 (complicated & mess) or simplers to just ditch react

Reply:
Does the esm-webpack-plugin work here? As I recall that's actually what you pointed out to me when I was doing work with the renderers needed to support module loading instead of script loading. We're already using it in our extension for that purpose.

Reply:
Actually nevermind, I'd prefer not to use react for something this simple anyways :).

Reply:
> oes the esm-webpack-plugin work here

No it doesn't, i think react has somethign special that makes it not work.

> d, I'd prefer not to use react for something this simple anyways :).

I agree, react is the new jquery, everyone seems to use that for anythign and everything to do with HTML...
The code isn't clean, but i think its readable enough and makes it smaller without react.